### DISCUSSION TITLE
Multi-language support on Shiny dashboard using existing crowd-sourcing platform

### DISCUSSION THREAD

**Discussion by joshwlambert** [WHO-Collaboratory/collaboratory-epipipeline-community#17] [Ideas]
WHO Collaboratory Hackathon Idea.

**Comment by ntncmch** [comment]
See R package shiny.i18n : https://appsilon.github.io/shiny.i18n/

**Comment by AmyMikhail** [comment]
This is something I've been looking into recently - have you come across [shi18y and friends](https://shi18ny.datasketch.dev/)?  At first glance it has the capacity to auto-translate common button labels and short text descriptors, but any more lengthy text (such as instructions for app users) would need to be done via a translation sheet.  This is possible with [shiny.i18n](https://github.com/Appsilon/shiny.i18n) but involves providing the translations yourself for the desired languages.  I wonder if this could be enhanced by using something like the [deeplr](https://github.com/zumbov2/deeplr) package, to get more automated translation for longer strings of text.

**Comment by AmyMikhail** [comment]
Also, slight tangent but somehow related - I just discovered the [countrycode](https://github.com/vincentarelbundock/countrycode) package, which has many variations of country names in its dictionary, in multiple languages and scripts. They also have names and codes used by various international entities - so far WHO regions are not on the list, but could easily be added.
